lib/galaxy/security/idencoding.py

```diff
@@ -99,8 +99,13 @@ def encode_guid(self, session_key):
 
     def decode_guid(self, session_key):
         # Session keys are strings
-        decoded_session_key = codecs.decode(session_key, 'hex')
-        return unicodify(self.id_cipher.decrypt(decoded_session_key)).lstrip('!')
+        try:
+            decoded_session_key = codecs.decode(session_key, "hex")
+            return unicodify(self.id_cipher.decrypt(decoded_session_key)).lstrip("!")
+        except TypeError:
+            raise galaxy.exceptions.MalformedId(f"Malformed guid '{session_key}' specified, unable to decode.")
+        except ValueError:
+            raise galaxy.exceptions.MalformedId(f"Wrong guid '{session_key}' specified, unable to decode.")
```

Comment:
If these 2 exception types come from `codecs.decode` and `decrypt()` respectively, maybe it's better to have 2 separate try/excepts.

Reply:
Maybe, but do we have to do this now? It's a bug fix that just uses what we know is working for the id decoding.